src/components/data-display/tables/UndernamesTable.tsx

```diff
@@ -184,53 +187,72 @@ const UndernamesTable = ({
                 className="flex flex-row justify-end pr-3 gap-3"
                 style={{ gap: '15px' }}
               >
-                <button
-                  onClick={() => {
-                    if (!arnsDomain || !antId) return;
-                    const targetName = encodePrimaryName(
-                      undername + '_' + arnsDomain,
-                    );
-                    if (primaryNameData?.name === targetName) {
-                      // remove primary name payload
-                      dispatchTransactionState({
-                        type: 'setTransactionData',
-                        payload: {
-                          names: [targetName],
-                          ioProcessId,
-                          assetId: antId,
-                          functionName: 'removePrimaryNames',
-                        },
-                      });
-                    } else {
-                      dispatchTransactionState({
-                        type: 'setTransactionData',
-                        payload: {
-                          name: targetName,
-                          ioProcessId,
-                          assetId: ioProcessId,
-                          functionName: 'primaryNameRequest',
-                        },
-                      });
-                    }
+                <Tooltip
```

Comment:
This code looks repeated in UndernamesSubtable. Would be nice to extract a reusable component and share between the two.